File: 🟢 Easy/125 - Valid Palindrome/_125ValidPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String str = s.toLowerCase();
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);
            if (ch >= 48 && ch <= 57) {
                sb.append(ch);
            }
            if (ch >= 97 && ch <= 122) {
                sb.append(ch);
            }

        }
        String fstr = sb.toString();
        String rev = sb.reverse().toString();
        if (fstr.equals(rev)) {
            return true;
        }
        return false;
    }
}

File: 🟢 Easy/125 - Valid Palindrome/_125ValidPalindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        for (auto& ch : s) {
            ch = tolower(ch);
        }
        
        string sb = "";

        for (int i = 0; i < s.length(); i++) {
            char ch = s.at(i);
            if (ch >= 48 && ch <= 57) {
                sb += (char)ch;
            }
            if (ch >= 97 && ch <= 122) {
                sb += (char)ch;
            }
        }

        string rev = sb;
        reverse(rev.begin(), rev.end());

        if (sb == rev) {
            return true;
        }

        return false;
    }
};